atom/renderer/api/atom_api_spell_check_client.h

```diff
@@ -85,6 +91,11 @@ class SpellCheckClient : public blink::WebSpellCheckPanelHostClient,
   SpellcheckWordIterator text_iterator_;
   SpellcheckWordIterator contraction_iterator_;
 
+  // The parameters of a pending background-spellchecking request.
+  // (When WebKit sends two or more requests, we cancel the previous
+  // requests so we do not have to use vectors.)
+  std::unique_ptr<SpellcheckRequest> pending_request_param_;
```

Comment:
I haven't read this patch deeply enough to say for sure, but maybe a `std::weak_ptr` would work better here?

Reply:
Maybe? my intention here is preserve chromium side logic as is if there isn't strong reason to replace (including comments), first it allows easier matching to chromium side implementation if anyone would like to compare, second it allows easier to pick up further upstream changes chromium does.

Reply:
Sounds good, keeping upstream's warts is reasonable IMO.